plugins/mysql/mysql_disk_by_prefix

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+# -*- python -*-
+
+"""
+=head1 INTRODUCTION
+
+Plugin to monitor MySQL database disk usage per prefix
+
+=head1 INSTALLATION
+
+Place in /etc/munin/plugins/ (or link it there using ln -s)
+
+=head1 CONFIGURATION
+
+Add this to your /etc/munin/plugin-conf.d/munin-node:
+
+=over 2
+
+    [mysql_disk_by_prefix]
+    user mysql
+    env.prefixes_with_underscores foo_bar d_ritchie # prefixes that include an underscore
+    env.db_minsize 1024000 # minimum db size in order to report a specific prefix,
+                           # in bytes (defaults to 50M). "0" for none.
+    env.mysql_db_dir /var/lib/mysql # default value
+
+=back
+
+=head1 HISTORY
+
+2019-07-25: v 1.0 pcy <pcy.ulyssis.org>: created
+
+=head1 USAGE
+
+Parameters understood:
+
+  config   (required)
+  autoconf (optional - used by munin-config)
+
+=head1 MAGIC MARKERS
+
+#%# family=auto
+#%# capabilities=autoconf
+"""
+
+
+import os
+import re
+import subprocess as sp
+import sys
+
+
+def weakbool(x):
+    return x.lower().strip() in {'true', 'yes', 'y', 1}
+
+
+LIMIT = os.getenv('db_minsize', str(50000 * 1024))
+try:
+    LIMIT = int(LIMIT)
+except Exception:
+    LIMIT = 0
+
+exceptions = os.getenv('prefix_with_underscores', '').split(' ')
+if exceptions == ['']:
+    exceptions = []
+
+mysqldir = os.getenv('mysql_db_dir', '/var/lib/mysql')
+
+
+def name_from_path(path):
+    filename = os.path.basename(path)
+    for name in exceptions:
+        if filename.startswith(name):
+            return name
+    name = filename.split('_')[0]
+
+    def decode_byte(m):
+        return bytes.fromhex(m.group(1)).decode('utf-8')
+
+    return re.sub('@00([0-9a-z]{2})', decode_byte, name)
```

Comment:
I assume, that this is mysql's approach of encoding non-ascii characters in database directory names? Maybe add a short comment mentioning this?

Reply:
Yep, commented that.